olive/engine/engine.py

```diff
@@ -406,6 +374,65 @@ def run(
 
         return outputs
 
+    def run_accelerator(
+        self,
+        input_model_config: ModelConfig,
+        data_root: str,
+        output_dir: str,
+        output_name: str,
+        evaluate_input_model: bool,
+        accelerator_spec: AcceleratorSpec,
+    ):
+        # generate search space and initialize the passes for each hardware accelerator
+        self.setup_passes(accelerator_spec)
+
+        # hash the input model
+        input_model_id = self._init_input_model(input_model_config)
+        self.footprints[accelerator_spec].record(model_id=input_model_id)
+
+        try:
+            if evaluate_input_model:
+                prefix_output_name = (
+                    f"{output_name}_{accelerator_spec}_" if output_name is not None else f"{accelerator_spec}"
+                )
+                assert self.evaluator_config is not None, "evaluate_input_model is True but no evaluator provided"
+                results = self._evaluate_model(
+                    input_model_config, input_model_id, data_root, self.evaluator_config, accelerator_spec
+                )
+                logger.info(f"Input model evaluation results: {results}")
+                result_name = f"{prefix_output_name}_input_model_metrics"
+                results_path = output_dir / f"{result_name}.json"
+                with open(results_path, "w") as f:
+                    json.dump(results.to_json(), f, indent=4)
+                logger.info(f"Saved evaluation results of input model to {results_path}")
+                if not self.passes:
+                    logger.debug("No passes registered, return input model evaluation results.")
+                    return results
```

Comment:
The results line L410, L413, L422 are different. Should we unify these output with same formatting?